cis194/week6/weining/src/Exercise5.hs

```diff
@@ -0,0 +1,27 @@
+module Exercise5
+  ( nats
+  , ruler
+  ) where
+
+import           Data.Bool
+import           Exercise4
+import           Stream
+
+nats :: Stream Integer
+nats = streamFromSeed (+ 1) 0
+
+{-
+the n-th element in the stream is the largest power of 2 which
+evenly divides n
+(assuming the first element corresponds to n=1)
+example:
+n 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16
+  0, 1, 0, 2, 0, 1, 0, 3, 0, 1, 0, 2, 0, 1, 0, 4
+-}
+ruler :: Stream Integer
+ruler = streamMap f $ streamFromSeed (+ 1) 1
+    where
+      f :: Integer -> Integer
```

Comment:
A note on naming of functions and variables. When we name polymorphic things, we use letter `x, y...`, `a, b`... to denote variable and `f, g, ...` to denote functions. In other cases, we want to give an expressive yet concise name.